modules/admin/src/main/java/org/motechproject/admin/internal/service/impl/ModuleAdminServiceImpl.java

```diff
@@ -263,6 +278,66 @@ public ExtendedBundleInformation getBundleDetails(long bundleId) {
         }
     }
 
+    private String parseDependencyVersion(Dependency dependency, MavenRepositorySystemSession mvnRepository, RepositorySystem system,
+                                          RemoteRepository remoteRepository, PomInformation pomInformation) throws ArtifactResolutionException {
+        String parsedVersion;
+        String version = dependency.getArtifact().getVersion();
+
+        if (StringUtils.isEmpty(version)) {
+            parsedVersion = "[0,)";
+        } else if (version.contains("${")) {
+            String propertyName = StringUtils.remove(version, "${");
```

Comment:
What if i I have ${baseVersion}-r1 as the version? or ${major.x.version}.${minor.x.version} ? There are better ways of parsing such tokens in Strings rather then doing it by hand.